distributed/protocol/serialize.py

```diff
@@ -440,6 +441,8 @@ def replace_inner(x):
 
 def serialize_bytelist(x, **kwargs):
     header, frames = serialize(x, **kwargs)
+    if "lengths" not in header:
```

Comment:
We (you :) ) recently added lengths to CUDA object headers.  When is lengths not in header ?  Is this something we should be requiring ?

Reply:
Yeah still wrapping my head around this. My understanding is the header from the object has already gone through msgpack at this point so is actually a frame as well. So it may be we always need to set the `length` here.